k2/csrc/host/intersect.cc

```diff
@@ -53,7 +53,7 @@ void Intersection::GetSizes(Array2Size<int32_t> *fsa_size) {
   arc_map_b_.clear();
 
   if (IsEmpty(a_) || IsEmpty(b_)) return;
-  if (!IsArcSorted(a_) || !IsArcSorted(b_)) {
+  if (check_properties_ && (!IsArcSorted(a_) || !IsArcSorted(b_))) {
     K2_LOG(WARNING) << "One of the inputs is not arc-sorted";
```

Comment:
This is for the original code. I think we can do better here to prevent unnecessary invoking of `IsArcSorted`.

```cpp
if (check_properties_) {
  if (!IsArcSorted(a_) {
    K2_LOG(INFO) << "Not arc-sorted: " << a_;
    status_ = false;
    return;
  }

  if (!IsArcSorted(b_) {
    K2_LOG(INFO) << "Not arc-sorted: " << b_;
    status_ = false;
    return;
  }
}
```

Reply:
And it is less error prone.